### PR TITLE
feat: redis server api support get command

### PR DESCRIPTION
**Pull request by liubaohai**
# Description           

impl redis api get

## Type of change       

- **New feature**       

# Checklist:

- [x] **Self-review**: I have performed a **self-review** of my own code
- [x] **Comment**:     I have **commented my code**, particularly in hard-to-understand areas
- [ ] **Doc**:         I have made corresponding changes to the **documentation**
- [x] **No-warnings**: My changes generate **no new warnings**
- [x] **Add-test**:    I have added **tests** that prove my fix is effective or that my feature works
- [x] **Pass**:        New and existing **unit tests pass** locally with my changes

### CODE REVIEW

File: src/redisapi/redisapi.rs [hunk 1]
```diff
@@ -138,54 +149,75 @@ impl RedisApi {
...
+                return Response::Error("invalid value".to_owned());
             }
         };
 
         let cmd = Command::from((cmd, key as &[u8], value as &[u8]));
         let cmds = vec![cmd];
 
-        let (g, r) = self.server_data.get_local_replica_for_key(key)?;
+        let (g, r) = try_with_err!(self.server_data.get_local_replica_for_key(key));
+        let mut st = try_with_err!(replicate(&cmds, g, r).await);
```
Comment (drmingdrmer) [inline src/redisapi/redisapi.rs:182]: hmmm...建议少用宏了, 这块如果能用Into处理的话感觉好看点

或者这块定义个cmd_xxx用的error, 然后还是返回result, 上层做into
转换成Response?

File: src/redisapi/redisapi.rs [hunk 2]
```diff
@@ -138,54 +149,75 @@ impl RedisApi {
...
+        try_with_err!(r.storage.set_instance(inst));
 
         // TODO bcast commit
+        let repl = try_with_err!(rx.await).pop().unwrap_or(None);
+        if let Some(v) = repl {
+            return Response::Data(v);
+        }
 
-        Ok(Response::Status("OK".to_owned()))
+        Response::Nil
```
Comment (drmingdrmer) [inline src/redisapi/redisapi.rs:221]: rx啥时候会返回None啊?

Comment (wenbobuaa) [reply src/redisapi/redisapi.rs:221]: key 不存在的时候？

Comment (liubaohai) [reply src/redisapi/redisapi.rs:221]: 对，key不存在的时候返回None

File: src/redisapi/redisapi.rs (hunk 1 above)
Comment (liubaohai) [reply src/redisapi/redisapi.rs:182]: 为啥这么排斥宏，有时候看起来挺简洁的

Comment (drmingdrmer) [reply src/redisapi/redisapi.rs]: 宏只有在作者自己看来简单, 因为作者明确知道它的作用和逻辑, 对不了解宏的定义的人来看有时必须把宏用脑子在阅读时展开才能看明白. 所以在中间逻辑里用宏感觉不如用函数看起来逻辑清楚. 有时候代码行数减少了, 但看的时候费劲了. 个人倾向于只在最末端的逻辑里用宏来减少重复, 最末端使用一些宏感觉不太会影响整体逻辑的形状. 🤔

**Pull request merged**